config/v1/config.go

```diff
@@ -125,12 +125,26 @@ type Module struct {
 	TerraformVersion *string `json:"terraform_version,omitempty" yaml:"terraform_version,omitempty"`
 }
 
+type CIProviderConfig struct {
+	Disabled bool `json:"disabled,omitempty" yaml:"disabled,omitempty"`
+}
+
+type CommonCI struct {
+	Enabled        *bool                       `json:"enabled,omitempty" yaml:"enabled,omitempty"`
+	AWSIAMRoleName *string                     `json:"aws_iam_role_name" yaml:"aws_iam_role_name,omitempty"`
+	TestBuckets    *int                        `json:"test_buckets" yaml:"test_buckets,omitempty"`
+	Command        *string                     `json:"command,omitempty" yaml:"command,omitempty"`
+	Buildevents    *bool                       `json:"buildevents,omitempty" yaml:"buildevents,omitempty"`
+	Providers      map[string]CIProviderConfig `json:"providers,omitempty" yaml:"providers,omitempty"`
+}
+
 type TravisCI struct {
-	Enabled        *bool   `json:"enabled,omitempty" yaml:"enabled,omitempty"`
-	AWSIAMRoleName *string `json:"aws_iam_role_name" yaml:"aws_iam_role_name,omitempty"`
-	TestBuckets    *int    `json:"test_buckets" yaml:"test_buckets,omitempty"`
-	Command        *string `json:"command,omitempty" yaml:"command,omitempty"`
-	Buildevents    *bool   `json:"buildevents,omitempty" yaml:"buildevents,omitempty"`
+	CommonCI `json:",inline" yaml:",inline"`
+}
+
+type CircleCI struct {
```

Comment:
Should we be adding CircleCI to the v1 config? It's a deprecated config format; it really should only be in v2.

Reply:
i'd rather be consistent with how we define travisCI rather than split these amongst packages

Reply:
Especially since they share several of the struct definitions

Reply:
Can't just move it to config/v2/config.go with a reference to v1.CommonCI? Personally, I don't like our v2 relying on v1 hack for Travis right now, but I'd rather not make it worse.

Reply:
> Personally, I don't like our v2 relying on v1 hack for Travis
I really don't see why it would be a hack. Would you rather duplicate the definitions?

Reply:
Made the changes